### PR TITLE
[1.2] Fold TaskDeploymentController into TaskExecutionController

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes https://github.com/spring-cloud/spring-cloud-dataflow/issues/1054

### CODE REVIEW

**Comment by cppwfs** [comment]
Need to update reference documentation to replace `tasks/deployments` with `tasks/executions`

**Comment by cppwfs** [comment]
Need story for UI once this is merged.  this is a breaking change for it.

**Comment by ericbottard** [comment]
> Need to update reference documentation to replace tasks/deployments with tasks/executions

This ought to be automatic, as we're using RESTDocs.

Story for UI could be done as part of https://github.com/spring-cloud/spring-cloud-dataflow/issues/1056 (cc @ghillert )

**Comment by cppwfs** [comment]
Index test in APIDocumentation.java needs to be updated.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskControllerTests.java [hunk 1]
```diff
@@ -204,20 +201,23 @@ public void testDestroyTaskNotFound() throws Exception {
...
+				post("/tasks/executions")
+					.param("name", "myTask")
+					.accept(MediaType.APPLICATION_JSON)).andDo(print())
 			.andExpect(status().is5xxServerError())
-			.andExpect(content().json("[{message: \"No URI found for task.nosuchtaskmodule\"}]"));
+			.andExpect(content().json("[{message: \"No URI found for task.no-such-task-app\"}]"));
 	}
 
 	@Test
 	public void testDeployNotDefined() throws Exception {
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskControllerTests.java]: rename taskNotDefined.

**Comment by ericbottard** [comment]
Rebased and addressed review comments

File: spring-cloud-dataflow-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java [hunk 2]
```diff
@@ -91,9 +91,7 @@ public void index() throws Exception {
 					linkWithRel("runtime/apps/instances").description("Provides the status for app instances"),
 					linkWithRel("tasks/definitions").description("Provides the task definition resource"),
 					linkWithRel("tasks/definitions/definition").description("Provides details for a specific task definition"),
-					linkWithRel("tasks/deployments").description("Provides the resource for deployment operations"),
-					linkWithRel("tasks/deployments/deployment").description("Launch a task"),
-					linkWithRel("tasks/executions").description("Returns Task executions"),
+					linkWithRel("tasks/executions").description("Returns Task executions and allows lanching of tasks"),
```
Comment (cppwfs) [inline spring-cloud-dataflow-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/ApiDocumentation.java]: launching is misspelled.  But will fix on merge.

**Comment by ericbottard** [comment]
Thanks.
Careful where you merge this though, master is still 1.1.1 for now. Maybe it's time to create the 1.1.x branch? cc @markfisher

**Comment by cppwfs** [comment]
Agreed.  I'm not merging till I see the 1.1.x branch show up.

**Comment by ericbottard** [comment]
Rebased

**Comment by ericbottard** [comment]
Rebased (once more)

**Pull request merged**